kahuna/public/js/search/results.js

```diff
@@ -170,12 +177,24 @@ results.controller('SearchResultsCtrl', [
             search({offset: start, length: length}).then(images => {
                 // Update imagesAll with newly loaded images
                 images.data.forEach((image, index) => {
-                    // Only set images that were missing from the Array
-                    // FIXME: might be safer to override, but causes
-                    // issues with object identity in the ng:repeat
-                    if (! ctrl.imagesAll[index + start]) {
-                        ctrl.imagesAll[index + start] = image;
+                    const position = index + start;
+                    const imageId = image.data.id;
+
+                    // If image already present in results at a
+                    // different position (result set shifted due to
+                    // items being spliced in or deleted?), get rid of
+                    // item at its previous position to avoid
+                    // duplicates
+                    const existingPosition = imagesPositions.get(imageId);
+                    if (angular.isDefined(existingPosition) &&
+                        existingPosition !== position) {
+                        $log.info(`Detected duplicate image ${imageId}, ` +
```

Comment:
Yes, this problem is such a pain I thought it's good to give a bit of info in case we need to debug in the wild.